### PR TITLE
change rlang abort to cli_abort

### LINKED ISSUE #1140: Use cli errors in `R/glmnet-engines.R`, `R/logistic_reg.R`, `R/mars.R`, and `R/mlp.R`
Most or all errors thrown in this package are made via `rlang::abort()`. We are transitioning to `cli::cli_abort()` to make use of the richer styling options for errors via cli (which calls rlang under the hood). This issue is part of https://github.com/tidymodels/parsnip/issues/1081

It is often sufficient to search for and replace the rlang version with the cli version together with adding a little bit of styling.

Some examples of the styling possibilities are in the [general introduction](https://cli.r-lib.org/articles/semantic-cli.html) under the section of [Inline text formatting](https://cli.r-lib.org/articles/semantic-cli.html#inline-text-formatting). More details on [inline formatting](https://cli.r-lib.org/reference/inline-markup.html) and [pluralization](https://cli.r-lib.org/articles/pluralization.html). 

One aspect to watch out for when transitioning from `rlang::abort()` to `cli::cli_abort()` are messages pre-assembled from user input as described [here](https://rlang.r-lib.org/dev/reference/topic-condition-formatting.html#transitioning-from-abort-to-cli-abort-).

#499 can serve as an example.

### PR DESCRIPTION
**Pull request by RobLBaker**
fixes #1140
@wright13

Co-author-by: Sarah Wright <sarah_wright@nps_gov>

### CODE REVIEW

File: R/glmnet-engines.R [hunk 1]
```diff
@@ -361,12 +361,13 @@ format_glmnet_multinom_class <- function(pred, penalty, lvl, n_obs) {
...
-      "To try multiple values for total regularization, use the tune package.",
-      "To predict multiple penalties, use `multi_predict()`"
-    ))
+    cli::cli_abort(c(
+      "x" = "For the glmnet engine, {.var penalty} must be a single number
+      (or a value of {.fn tune}).",
+      "!" = "There are {length(pen)} value{?s} for {.var penalty}.",
+      "i" = "To try multiple values for total regularization, use the
+      {.pkg tune} package.",
+      "i" = "To predict multiple penalties, use {.fn multi_predict}."))
```
Comment (simonpcouch) [inline R/glmnet-engines.R:370]:
```suggestion
      "i" = "To predict multiple penalties, use {.fn multi_predict}."
    ))
```

File: R/glmnet-engines.R [hunk 2]
```diff
@@ -385,23 +386,19 @@ format_glmnet_multinom_class <- function(pred, penalty, lvl, n_obs) {
...
     if (length(penalty) != 1) {
-      rlang::abort(
-        glue::glue(
-          "`penalty` should be a single numeric value. `multi_predict()` ",
-          "can be used to get multiple predictions per row of data."
-        )
-      )
+      cli::cli_abort(c(
+        "{.var penalty} should be a single numeric value. {.fn multi_predict}
+        can be used to get multiple predictions per row of data."
```
Comment (simonpcouch) [inline R/glmnet-engines.R:391]:
```suggestion
        "{.var penalty} should be a single numeric value.",
        "i" = "{.fn multi_predict} can be used to get multiple predictions per row of data."
```

File: R/glmnet-engines.R [hunk 3]
```diff
@@ -385,23 +386,19 @@ format_glmnet_multinom_class <- function(pred, penalty, lvl, n_obs) {
...
-    rlang::abort(
-      glue::glue(
-        "The glmnet model was fit with a single penalty value of ",
-        "{object$fit$lambda}. Predicting with a value of {penalty} ",
-        "will give incorrect results from `glmnet()`."
-      )
-    )
+    cli::cli_abort(c(
+      "The glmnet model was fit with a single penalty value of
+      {.var object$fit$lambda}. Predicting with a value of {.envar penalty}
```
Comment (simonpcouch) [inline R/glmnet-engines.R:399]:
```suggestion
      {.arg object$fit$lambda}. Predicting with a value of {.arg penalty}
```

Not the fault of this PR (so feel free to ignore), but I find this message a bit unclear

File: R/logistic_reg.R [hunk 4]
```diff
@@ -98,7 +98,7 @@ translate.logistic_reg <- function(x, engine = x$engine, ...) {
         } else if (quo_get_expr(x$args$mixture) == 1) {
           arg_vals$type <- 6      ## lasso
         } else {
-          rlang::abort("For the LiblineaR engine, mixture must be 0 or 1.")
+          cli::cli_abort(c("x" = "For the LiblineaR engine, {.var mixture} must be 0 or 1."))
```
Comment (simonpcouch) [inline R/logistic_reg.R:101]:
```suggestion
          cli::cli_abort("For the LiblineaR engine, {.var mixture} must be 0 or 1.")
```

`cli_abort()` will automatically prefix the first element of the message with an icon!

File: R/mars.R [hunk 5]
```diff
@@ -161,14 +161,14 @@ multi_predict._earth <-
         object$fit$call[[i]] <- eval_tidy(object$fit$call[[i]])
     }
 
-    msg <-
-      paste("Please use `keepxy = TRUE` as an option to enable submodel",
-            "predictions with `earth`.")
+    msg <- c("x" = "Please use {.code keepxy = TRUE} as an
+                            option to enable submodel predictions with
+                            {.var earth}.")
```
Comment (simonpcouch) [inline R/mars.R:166]:
```suggestion
    msg <- 
      c("x" = "Please use {.code keepxy = TRUE} as an option to enable submodel 
                     predictions with earth.")
```

File: R/mlp.R [hunk 6]
```diff
@@ -148,7 +148,7 @@ check_args.mlp <- function(object, call = rlang::caller_env()) {
 
 class2ind <- function (x, drop2nd = FALSE) {
   if (!is.factor(x))
-    rlang::abort("`x` should be a factor")
+    cli::cli_abort(c("x" = "{.var x} should be a factor."))
```
Comment (simonpcouch) [inline R/mlp.R:151]:
```suggestion
    cli::cli_abort(c("x" = "{.arg x} should be a factor."))
```

File: R/mlp.R [hunk 7]
```diff
@@ -195,8 +195,9 @@ keras_mlp <-
     rlang::arg_match(activation, act_funs,)
 
     if (penalty > 0 & dropout > 0) {
-      rlang::abort("Please use either dropoput or weight decay.", call. = FALSE)
-    }
+      cli::cli_abort(c("x" = "Please use either dropout or weight decay",
+                       call = NULL))
```
Comment (simonpcouch) [inline R/mlp.R:199]:
```suggestion
      cli::cli_abort("Please use either dropout or weight decay.", call = NULL)
```

File: R/glmnet-engines.R [hunk 8]
```diff
@@ -361,12 +361,13 @@ format_glmnet_multinom_class <- function(pred, penalty, lvl, n_obs) {
...
 
   if (length(pen) != 1) {
-    rlang::abort(c(
-      "For the glmnet engine, `penalty` must be a single number (or a value of `tune()`).",
-      glue::glue("There are {length(pen)} values for `penalty`."),
-      "To try multiple values for total regularization, use the tune package.",
-      "To predict multiple penalties, use `multi_predict()`"
-    ))
+    cli::cli_abort(c(
+      "x" = "For the glmnet engine, {.var penalty} must be a single number
```
Comment (simonpcouch) [inline R/glmnet-engines.R:365]:
```suggestion
      "x" = "For the glmnet engine, {.arg penalty} must be a single number
```

**Review by simonpcouch — COMMENTED** [review]
A great start! Thanks for yall's work.

File: R/logistic_reg.R [hunk 9]
```diff
@@ -98,7 +98,7 @@ translate.logistic_reg <- function(x, engine = x$engine, ...) {
         } else if (quo_get_expr(x$args$mixture) == 1) {
           arg_vals$type <- 6      ## lasso
         } else {
-          rlang::abort("For the LiblineaR engine, mixture must be 0 or 1.")
+          cli::cli_abort("For the LiblineaR engine, {.var mixture} must be 0 or 1.")
```
Comment (simonpcouch) [inline R/logistic_reg.R:101]:
```suggestion
          cli::cli_abort("For the LiblineaR engine, {.arg mixture} must be 0 or 1.")
```

File: R/mlp.R [hunk 10]
```diff
@@ -195,8 +195,8 @@ keras_mlp <-
     rlang::arg_match(activation, act_funs,)
 
     if (penalty > 0 & dropout > 0) {
-      rlang::abort("Please use either dropoput or weight decay.", call. = FALSE)
-    }
+      cli::cli_abort("Please use either dropout or weight decay.", call = NULL)
+      }
```
Comment (simonpcouch) [inline R/mlp.R:199]:
```suggestion
    }
```

**Review by simonpcouch — APPROVED** [review]
Stellar.🔨

**Comment by simonpcouch** [comment]
Thank you both!

**Comment by RobLBaker** [comment]
thanks for the detailed comments and styling help. Hoping to find more of these to pitch in on.

**Pull request merged**